pkg/chunk/cached_store.go

```diff
@@ -289,10 +288,20 @@ func (c *rChunk) Remove() error {
 		delete(c.store.pendingKeys, key)
 		c.store.pendingMutex.Unlock()
 		c.store.bcache.remove(key)
+
+	}
+
+	if c.store.conf.MaxDeletes == 0 {
+		return nil
+	}
+	var err error
+	c.store.currentDelete <- struct{}{}
```

Comment:
There could be too many goroutines pending for deletions.

Reply:
btw, Same problem as before this change

Reply:
should fix in another PR